app/controllers/messages_controller.rb

```diff
@@ -73,7 +73,7 @@ def set_conversation
   end
 
   def set_assistant
-    @assistant = Current.user.assistants_including_deleted.find_by(id: params[:assistant_id])
+    @assistant = Current.user.assistants_including_deleted.find_by(slug: params[:assistant_id])
```

Comment:
I was a bit surprised that this worked because I did not see you change a corresponding link_to in order to generate a url that contains this slug. Is that some default rails helper? I tried asking gpt4 but it claimed not so I don't get how the URLs are now using slug rather than id.

Reply:
See `Assistant#to_param` which is magically used by rails routes

Reply:
Oh, clever! But now I can't figure out why a newly added assistant is not working? Based on that implementation I would expect it to recognize any slug. Let's move over to this thread: https://github.com/AllYourBot/hostedgpt/pull/525#issuecomment-2505145760